core/src/test/java/com/linecorp/armeria/common/RequestContextTest.java

```diff
@@ -307,6 +308,83 @@ public void makeContextAwareCompletableFutureWithAsyncChaining() throws Exceptio
         }
     }
 
+    @SuppressWarnings("unchecked")
+    @Test
+    public void minimalCompletionStageUsingCompleteAsync() throws Exception {
+        final RequestContext context = createContext(false);
+        final CompletableFuture<Integer> originalFuture = new CompletableFuture<>();
+        final RequestContextAwareCompletableFuture<Integer> contextAwareFuture =
+                (RequestContextAwareCompletableFuture) context.makeContextAware(originalFuture);
+        final CompletionStage<Integer> completionStage = contextAwareFuture.minimalCompletionStage();
+
+        contextAwareFuture.completeAsync(() -> {
+            throw new RuntimeException();
+        });
+        contextAwareFuture.complete(1);
+
+        assertThat(contextAwareFuture.join()).isEqualTo(1);
+        assertThat(contextAwareFuture.getNow(null)).isEqualTo(1);
+        assertThat(contextAwareFuture.get()).isEqualTo(1);
+        completionStage.toCompletableFuture().get();
+    }
+
+    @SuppressWarnings("unchecked")
```

Comment:
You do not need to add this annotation at the method level but limit the scope of suppression at variable level:

```java
@SuppressWarnings("unchecked")
final RequestContextAwareCompletableFuture<Integer> contextAwareFuture =
        (RequestContextAwareCompletableFuture) context.makeContextAware(originalFuture);
```